packages/fxa-react/lib/utils.tsx

```diff
@@ -37,10 +41,8 @@ export function hardNavigate(
     });
   }
 
-  if (href.includes('?')) {
-    href = href.substring(0, href.indexOf('?'));
-  }
-  window.location.href = `${href}?${searchParams.toString()}`;
+  window.location.href =
+    searchParams.size > 0 ? `${path}?${searchParams.toString()}` : path;
```

Comment:
Looks fine, but I think all the changes in this file would be simpler if we construct a `URL` from the `href`.

Reply:
You're so right! Fresh eyes on this is very helpful. Please check out:
https://github.com/mozilla/fxa/pull/17017/commits/26bdedb5b02cc096637d4742a7c8f510d8fb3ecf